Add support for inline classes

DanTup opened the pull request:
See https://github.com/dart-lang/sdk/issues/49734 and https://github.com/dart-lang/language/blob/master/accepted/future-releases/inline-classes/feature-specification.md.

Inline is only detected as part of class, since the spec says:

> The token `inline` is not made a built-in identifier: the reserved word `class` that occurs right after `inline` serves to disambiguate the inline class declaration with a fixed lookahead.

kevmoo commented:
Thanks, @DanTup !

devoncarew reviewed (approved):
🎉

devoncarew commented:
@DanTup - can you resolve the change conflict here? And, you now have write access to this repo, so will be able to merge your own PRs (after review).

DanTup commented:
@devoncarew thanks! I've rebased and updated the version/date. I presume it's reasonable to land this now even if inline classes may be a little way off (it's just the single keyword)?

devoncarew commented:
> I presume it's reasonable to land this now even if inline classes may be a little way off (it's just the single keyword)?

Ah, good question. It may be better to hold off on that until more things start landing / we have a better sense of how far out it is.

DanTup commented:
sgtm, I'll leave it here for now and update the version/date as appropriate when things might be clearer. Thanks!

DanTup commented:
@devoncarew @jwren based on discussions it sounds like it's reasonable to merge this?

I've updated the date in changelog, though the version is still the same as nothing else has been committed since I first did this. Let me know if you've happy for me to merge (seems like I have merge access here now 👍).

jwren commented:
Yes, do merge, if there are any other syntax changes we can always follow up.
